lib/util/LazySet.js

```diff
@@ -101,12 +101,14 @@ class LazySet {
 				_set.add(item);
 			}
 		} else {
-			this._toDeepMerge.push(iterable);
-			this._needMerge = true;
-			// Avoid being too memory hungry
-			if (this._toDeepMerge.length > 100000) {
-				this._flatten();
-				if (this._toMerge.size > 100000) this._merge();
+			if (!(iterable instanceof LazySet) || iterable.size) {
```

Comment:
Asking a LazySet for it's size will cause a `merge` of it, which makes this operation no longer lazy. Best add a new method to ask for `isEmpty`

Reply:
When checking for instanceof LazySet here I would like to avoid checking again in flatten.

Could you put only non-empty LazySets into toDeepMerge?
Other things should go directly into toMerge.
Check for overflow in the set you modified.

Reply:
Yea, I noticed this but found perf still improved when making it lazy-ish. Anyway, I have incorporated your suggestions and added a few more test cases. Thanks for the review @sokra